public/docs/ts/latest/cookbook/aot-compiler.jade

```diff
@@ -0,0 +1,177 @@
+
+    Since `NgFactories` are generated, we should not make any edits to them since any edits will be lost the next time we AOT compile the application. 
+
+    With JIT compilation the factory files are out of sight, out of mind. The same `NgFactory` files are generated during JIT compilation though. We just never see the physical files because it happens at runtime - on the fly. 
+
+<a id="compilation"></a>
+:marked
+    ## Tree Shaking
+:marked
+    AOT compilation also sets the stage for further optimization from a process called Tree Shaking. Tree Shaking means walking our code dependency tree, top to bottom, and shake out any unused code.
+
+    Tree Shaking is great for reducing the download size of our application since it effectively removes any unused portions of our source code. 
+    
+    Tree Shaking even applies to the Angular framework code. In fact, most of our size reduction will likely come from removing unreferenced features in the Angular code. 
+    
+    As an example, if our application is not using any `Forms` functionality, there is no reason to make the end user download any Forms related Angular code.
+
+    Three Shaking and AOT compilation are separate steps, but Tree Shaking can only target JavaScript code. By doing AOT compilation we convert more of our application to JavaScript, which in turn makes more of our application "Tree Shakable".
+
+    ### Rollup
+
+    In our example we are using a framework called Rollup to do the Tree Shaking.
+
+    Rollup works best with `ES2015` source code, so we transpile our application to `ES2015` by setting the `target` and `module` properties in `tsconfig.json` to `es2015`. This takes care of onverting our TypeScript code to `ES2015` JavaScript. 
+    
+    We also specify an output folder for `NgFactories` in `tsconfig.json` by defining `angularCompilerOptions` with a `genDir`. 
+
+    Next we have to create a configuration file to tell Rollup how to process our application. 
+    
+    In our example we have named the configuration file rollup.js.
+
+    In rollup.js we tell Rollup that the entry point to our application is app/main.js by setting the `entry` attribute.
+
+    We want Rollup to output a single application bundle containing our application code, and only the parts of Angular (and other dependencies) that we actually use. 
+    
+    By setting the `dest` attribute we tell Rollup where to put the generated bundle. In this example it will be generated as dist/build.js.
+
++makeTabs(
+  `cb-ahead-of-time-compilation/ts/rollup.js,
+   cb-ahead-of-time-compilation/ts/app/tsconfig.json`,
+  null,
+  `rollup.js,
+   tsconfig.json`
+) 
+
+:marked
+    ### Rollup Plugins
+
+    Our application code is now generated in `ES2015` format and the Angular framework is by default deployed as `ES2015`. However, we have one slight challenge with RxJs. 
+    
+    Angular has a direct dependency on RxJs, but RxJs code bundles are not published as `ES2015` JavaScript.
```

Comment:
This is something I know @wardbell has commented on in the past. The phrasing of the RxJS dependency may be something he wants to articulate.